influxql/ast.go

```diff
@@ -1965,24 +1989,56 @@ type Visitor interface {
 
 // Walk traverses a node hierarchy in depth-first order.
 func Walk(v Visitor, node Node) {
+	if node == nil {
```

Comment:
An optional section in a statement might be represented by a pointer.  A `nil` pointer indicates that the section is not present.  Checking at the top of the `Walk` func prevents each case statement having to check for `nil` on optional sections.  It can safely pass `nil` when it makes the recursive call to itself.